dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/JhoveConfigurationInitializer.java

```diff
@@ -1,71 +1,85 @@
 package edu.harvard.iq.dataverse.util;
 
-import org.apache.commons.io.FileUtils;
-import org.apache.commons.lang3.StringUtils;
+import static java.lang.System.getProperty;
+import static java.lang.Thread.currentThread;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.Files.copy;
+import static java.nio.file.Files.newBufferedWriter;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static org.apache.commons.io.IOUtils.resourceToString;
+import static org.apache.commons.lang3.StringUtils.replaceOnce;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Writer;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
 import javax.annotation.PostConstruct;
-import javax.annotation.PreDestroy;
 import javax.ejb.Singleton;
 import javax.ejb.Startup;
 
-import java.io.ByteArrayInputStream;
-import java.io.File;
-import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.nio.file.StandardCopyOption;
-
 /**
  * Initializer of jhove configuration files.
  * 
- * Classes that uses jhove should assume that
- * configuration file for jhove is located in
- * {@link #JHOVE_CONFIG_PATH}.
+ * Classes that uses jhove should assume that configuration file for jhove is
+ * located in {@link #getConfigPath}.
  */
 @Startup
 @Singleton
 public class JhoveConfigurationInitializer {
+	
+	static Path getConfigPath() {
+		
+		return pathOf("jhove.conf");
+	}
+	
+	/**
+	 * Copies jhove configuration and configuration xsd files from classpath into
+	 * temp directory. In case of configuration file - it also replaces xsd location
+	 * placeholder to the path in temp directory where it was copied.
+	 * 
+	 * Replacing of schema location is done, because jhove will validate given
+	 * configuration file against it.
+	 */
+	@PostConstruct
+	public void initializeJhoveConfig() {
+
+		try {
+			final Path xsd = pathOf("jhoveConfig.xsd");
+			copyXsd(xsd);
+			copyConfig(xsd);
```

Comment:
Nit: The meaning of the `xsd` variable changes in the above two calls. In the first it's kind of an output parameter and an input in the 2nd one.
To make it more clearer one could let `copyXsd` method return the path to the resulting xsd file instead of receiving it as parameter. Alternatively one could also rename the methods, like `copyXsdTo(xsdPath)` and `createConfigWith(xsdPath)`.

Reply:
fixed